File: backend/Minimarket.Api/Controllers/PrintJobsController.cs
using Microsoft.AspNetCore.Mvc;
using Minimarket.Api.DTOs;
using Minimarket.Api.Services;

namespace Minimarket.Api.Controllers;

[ApiController]
[Route("api/[controller]")]
public class PrintJobsController(IPrintJobService printJobService) : ControllerBase
{
    [HttpGet("recent")]
    public async Task<ActionResult<IReadOnlyCollection<PrintJobDto>>> GetRecent([FromQuery] int take = 20) =>
        Ok(await printJobService.GetRecentAsync(Math.Clamp(take, 1, 100)));

    [HttpPost("sales/{saleId:int}/enqueue")]
    public async Task<ActionResult<PrintJobDto>> EnqueueSaleTicket(int saleId)
    {
        var result = await printJobService.EnqueueSaleTicketAsync(saleId);
        if (!result.Success)
        {
            return BadRequest(new { message = result.Error });
        }

        return Ok(result.Job);
    }

    [HttpPost("{id:int}/requeue")]
    public async Task<ActionResult<PrintJobDto>> Requeue(int id)
    {
        var result = await printJobService.RequeueAsync(id);
        if (!result.Success)
        {
            return BadRequest(new { message = result.Error });
        }

        return Ok(result.Job);
    }
}

File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/controller/PrintJobsController.java
package com.minimarket.api.controller;

import com.minimarket.api.dto.ApiMessageResponse;
import com.minimarket.api.dto.PrintJobDto;
import com.minimarket.api.service.PrintJobService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/print-jobs")
public class PrintJobsController {

    private final PrintJobService printJobService;

    public PrintJobsController(PrintJobService printJobService) {
        this.printJobService = printJobService;
    }

    @GetMapping("/recent")
    public List<PrintJobDto> getRecent() {
        return printJobService.getRecent();
    }

    @PostMapping("/sales/{saleId}/enqueue")
    public ResponseEntity<?> enqueueSaleTicket(@PathVariable Integer saleId) {
        var result = printJobService.enqueueSaleTicket(saleId);
        if (!result.success()) {
            return ResponseEntity.badRequest().body(new ApiMessageResponse(result.error()));
        }

        return ResponseEntity.ok(result.data());
    }

    @PostMapping("/{id}/requeue")
    public ResponseEntity<?> requeue(@PathVariable Integer id) {
        var result = printJobService.requeue(id);
        if (!result.success()) {
            return ResponseEntity.badRequest().body(new ApiMessageResponse(result.error()));
        }

        return ResponseEntity.ok(result.data());
    }
}
